lib/puppet/type/zpool.rb

```diff
@@ -90,6 +90,22 @@ def insync?(is)
       desc 'Log disks for this pool. This type does not currently support mirroring of log disks.'
     end
 
+    newproperty(:ashift) do
+      desc 'The Alignment Shift for the vdevs in the given pool.'
+
+      validate do |_value|
+        raise Puppet::Error _('This property is only supported on Linux') unless Facter.value(:kernel) == 'Linux'
```

Comment:
I've been using `-o ashift=12` on FreeBSD.  I suspect it should be usable there too.

Reply:
I'll update this to include FreeBSD

Reply:
Done